src/structure/model.jl

```diff
@@ -222,378 +299,574 @@ end
 #  residue ID
-Base.getindex(chain::Chain, res_id::AbstractString) = chain.residues[res_id]
-function Base.setindex!(chain::Chain, res::AbstractResidue, res_id::AbstractString)
-    chain.residues[res_id] = res
-    return chain
+Base.getindex(ch::Chain, res_id::AbstractString) = ch.residues[res_id]
+
+function Base.setindex!(ch::Chain, res::AbstractResidue, res_id::AbstractString)
+    ch.residues[res_id] = res
+    return ch
 end
 
 # Accessing a Chain with an Integer returns the AbstractResidue with that residue ID
 #   converted to a string
-Base.getindex(chain::Chain, res_number::Integer) = chain.residues[string(res_number)]
-function Base.setindex!(chain::Chain, res::AbstractResidue, res_number::Integer)
-    chain.residues[string(res_number)] = res
-    return chain
+Base.getindex(ch::Chain, res_n::Integer) = ch.residues[string(res_n)]
+
+function Base.setindex!(ch::Chain, res::AbstractResidue, res_n::Integer)
+    ch.residues[string(res_n)] = res
+    return ch
 end
 
 # Accessing a Model with a Char returns the Chain with that chain ID
-Base.getindex(model::Model, chain_id::Char) = model.chains[chain_id]
-function Base.setindex!(model::Model, chain::Chain, chain_id::Char)
-    model.chains[chain_id] = chain
-    return model
+Base.getindex(mod::Model, ch_id::Char) = mod.chains[ch_id]
+
+function Base.setindex!(mod::Model, ch::Chain, ch_id::Char)
+    mod.chains[ch_id] = ch
+    return mod
 end
 
-# Accessing a ProteinStructure with an Integer returns the Model with that model
+# Accessing a ProteinStructure with an Integer returns the Model with that model
 #   number
-Base.getindex(struc::ProteinStructure, model_number::Integer) = struc.models[model_number]
-function Base.setindex!(struc::ProteinStructure, model::Model, model_number::Integer)
-    struc.models[model_number] = model
+Base.getindex(struc::ProteinStructure, mod_n::Integer) = struc.models[mod_n]
+
+function Base.setindex!(struc::ProteinStructure, mod::Model, mod_n::Integer)
+    struc.models[mod_n] = mod
     return struc
 end
 
 # Accessing a ProteinStructure with a Char returns the Chain with that chain ID
 #   on the default model
-Base.getindex(struc::ProteinStructure, chain_id::Char) = defaultmodel(struc)[chain_id]
-function Base.setindex!(struc::ProteinStructure, chain::Chain, chain_id::Char)
-    defaultmodel(struc)[chain_id] = chain
+Base.getindex(struc::ProteinStructure, ch_id::Char) = defaultmodel(struc)[ch_id]
+
+function Base.setindex!(struc::ProteinStructure, ch::Chain, ch_id::Char)
+    defaultmodel(struc)[ch_id] = ch
     return struc
 end
 
+# Check if an atom name exists in a residue as a whitespace-padded version
+function findatombyname(res::Residue, atom_name::AbstractString)
+    # Look for atom name directly
+    if atom_name in keys(res.atoms)
+        return res.atoms[atom_name]
+    # Pad out name to 4 characters to read PDB atom names with whitespace
+    elseif length(atom_name) == 3
```

Comment:
I'm not sure but I thought that linear search may be faster than padding names because this padding will create so many garbage objects while searching.